types/controller/forwarding/internal.go

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, NASD Inc. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN "AS IS" BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package forwarding
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/noble-assets/orbiter/types/core"
+)
+
+var _ core.ForwardingAttributes = &InternalAttributes{}
+
+// CounterpartyID implements core.ForwardingAttributes.
+func (a *InternalAttributes) CounterpartyID() string {
+	return "noble"
+}
+
+func NewInternalAttributes(
+	recipient string,
+) (*InternalAttributes, error) {
+	attr := InternalAttributes{
+		Recipient: recipient,
+	}
+
+	return &attr, attr.Validate()
+}
+
+func (a *InternalAttributes) Validate() error {
+	if a == nil {
+		return core.ErrNilPointer.Wrap("internal attributes")
+	}
+
+	if a.Recipient == "" {
+		return errorsmod.Wrapf(core.ErrEmptyString, "invalid recipient address")
+	}
+
+	if _, err := sdk.AccAddressFromBech32(a.Recipient); err != nil {
+		return errorsmod.Wrapf(err, "invalid recipient address")
+	}
```

Comment:
nit, it's a redundant check because `""` would also fail the `AccAddressFromBech32` function. We can also leave it as is but I think just having the address conversion check is fine.


```suggestion
	if _, err := sdk.AccAddressFromBech32(a.Recipient); err != nil {
		return errorsmod.Wrapf(err, "invalid recipient address")
	}
```